packages/frontend/src/connectors/readonly/index.ts

```diff
@@ -2,14 +2,43 @@ import { JsonRpcProvider } from "@ethersproject/providers";
 import { Connector } from "wagmi";
 import { Address, Chain, ConnectorData, normalizeChainId } from "@wagmi/core";
 import { constants } from "ethers";
+import { ChainId } from "@carrot-kpi/sdk";
 
 export const READ_ONLY_CONNECTOR_ID = "readonly";
+const DEFAULT_CHAIN_ID = ChainId.GNOSIS;
 
 export type InjectedConnectorOptions = {
     /** Name of connector */
     name?: string | ((detectedName: string | string[]) => string);
 };
 
+interface wagmiStoreObj {
```

Comment:
We can change this to `WagmiStore` (also with capital `W`).
I think `state, data, chain, id` should all be optional with `?`, since they may be `undefined`.

Reply:
fixed: bce2c26da9eea0de96637b3dfa30bee94e97c732